backend/src/services/premium/enrichment/organizationEnrichmentService.ts

```diff
@@ -0,0 +1,216 @@
+      this.log.error({ err }, 'Error in api-ws emitter!')
+    })
+    if (!organizations.length) {
+      apiPubSubEmitter.emit(
+        'user',
+        new ApiWebsocketMessage(
+          'organization-bulk-enrichment',
+          JSON.stringify({
+            tenantId: this.tenantId,
+            success: false,
+            organizationIds,
+          }),
+          undefined,
+          this.tenantId,
+        ),
+      )
+    }
+    // Send success message if there were enrichedMembers
+    else {
+      apiPubSubEmitter.emit(
+        'user',
+        new ApiWebsocketMessage(
+          'organization-bulk-enrichment',
+          JSON.stringify({
+            tenantId: this.tenantId,
+            success: true,
+            organizationIds,
+          }),
+          undefined,
+          this.tenantId,
+        ),
+      )
+    }
+  }
+
+  private selectFieldsForEnrichment(org: IEnrichableOrganization): IEnrichableOrganization {
+    this.fields.push('name', 'lastEnrichedAt')
+    if(OrganizationEnrichmentService.shouldReenrich(org)) {
+      return org
+    }
+    for(const field of Object.keys(org)) {
+      if(org[field] === null){
+        this.fields.push(field)
+      }
+    }
+    return lodash.pick(org, ['id', 'cachId', ...this.fields])
+  }
+
+  async queryTenancyOrganizations(): Promise<IEnrichableOrganization[]> {
+    const options = await SequelizeRepository.getDefaultIRepositoryOptions()
+    const query = `
+      with orgActivities as (
+        SELECT memOrgs."organizationId", SUM(actAgg."activityCount") "orgActivityCount"
+        FROM "memberActivityAggregatesMVs" actAgg
+        INNER JOIN "memberOrganizations" memOrgs ON actAgg."id"=memOrgs."memberId"
+        GROUP BY memOrgs."organizationId"
+      ) 
+      SELECT org.id "id"
+      ,cach.id "cachId"
+      ,org."name"
+      ,org."location"
+      ,org."website"
+      ,org."lastEnrichedAt"
+      ,org."twitter"
+      ,org."employees"
+      ,org."size"
+      ,org."founded"
+      ,org."industry"
+      ,org."naics"
+      ,org."profiles"
+      ,org."headline"
+      ,org."ticker"
+      ,org."type"
+      ,org."employeeCountByCountry"
+      ,org."description"
+      FROM "organizations" as org
+      JOIN "organizationCaches" cach ON org."name" = cach."name"
+      JOIN orgActivities activity ON activity."organizationId" = org."id"
+      WHERE :tenantId = org."tenantId" AND (org."lastEnrichedAt" IS NULL OR org."lastEnrichedAt" >= NOW() - INTERVAL '6 months')
+      ORDER BY org."lastEnrichedAt" ASC, activity."orgActivityCount" DESC, org."createdAt" DESC
+      LIMIT :limit
+    ;
+    `
+    const orgs: IEnrichableOrganization[] = await SequelizeRepository.getSequelize(options).query(
+      query,
+      {
+        type: QueryTypes.SELECT,
+        replacements: {
+          tenantId: this.tenantId,
+          limit: this.maxOrganizationsLimit,
+        },
+      },
+    )
+    return orgs.map(org => this.selectFieldsForEnrichment(org))
```

Comment:
From [PDL's docs](https://docs.peopledatalabs.com/docs/company-stats) it looks like we obtain:
- `employee_count` (we should have this as `employees`)
- `location`: it's an object with several interesting fields. I think we should store all of them as a JSONB. Except for the GEO code. Let's make a new column for it.

Reply:
We are not currently updating `employees` with `employee_count` from PDL as it's not in the task requirement document.